perf_test/sparse/KokkosSparse_sptrsv_aux.hpp

```diff
@@ -0,0 +1,626 @@
+                    int *perm_r, int *perm_c, double tol, int loop) {
+#if defined(KOKKOSKERNELS_ENABLE_TPL_CUSPARSE) 
+  using values_view_t = typename crsmat_t::values_type::non_const_type;
+  using scalar_t      = typename values_view_t::value_type;
+
+  using host_values_view_t = typename host_crsmat_t::values_type::non_const_type;
+  using host_scalar_t      = typename host_values_view_t::value_type;
+
+  using execution_space = typename values_view_t::execution_space;
+  using memory_space    = typename execution_space::memory_space;
+
+  using host_execution_space = typename host_values_view_t::execution_space;
+  using host_memory_space    = typename host_execution_space::memory_space;
+
+  using host_scalar_view_t = Kokkos::View< scalar_t*, host_memory_space >;
+  using      scalar_view_t = Kokkos::View< scalar_t*,      memory_space >;
+
+  scalar_t ZERO = scalar_t (0);
+  scalar_t ONE = scalar_t (1);
+
+  Kokkos::Timer timer;
+  const int nrows = Mtx.graph.numRows ();
+
+  // ==============================================
+  // > create a handle
+  cusparseStatus_t status;
+  cusparseHandle_t handle = 0; 
+  status = cusparseCreate (&handle);
+  if (CUSPARSE_STATUS_SUCCESS != status) {
+    std::cout << "handle create status error name " << (status) << std::endl;
+  }
+  cusparseSetPointerMode (handle, CUSPARSE_POINTER_MODE_HOST); // scalars are passed by reference on host
+
+  // > create a empty info structure for L-solve (e.g., analysis results)
+  csrsv2Info_t infoL = 0; 
+  status = cusparseCreateCsrsv2Info (&infoL);
+  if (CUSPARSE_STATUS_SUCCESS != status) {
+    std::cout << "csrsv2info create status error name " << (status) << std::endl;
+  }
+
+  // ==============================================
+  // Preparing for L-solve
+  // step 1: create a descriptor
+  int nnzL = L.nnz ();
+  auto graphL = L.graph; // in_graph
+  auto row_mapL = graphL.row_map;
+  auto entriesL = graphL.entries;
+  auto valuesL  = L.values;
+
+  // NOTE: it is stored in CSC = UPPER + TRANSPOSE
+  cusparseMatDescr_t descrL = 0;
+  status = cusparseCreateMatDescr (&descrL);
+  if (CUSPARSE_STATUS_SUCCESS != status) {
+    std::cout << "matdescr create status error name " << (status) << std::endl;
+  }
+  cusparseSetMatIndexBase (descrL, CUSPARSE_INDEX_BASE_ZERO);
+  cusparseSetMatFillMode (descrL, CUSPARSE_FILL_MODE_UPPER);
+  cusparseSetMatDiagType (descrL, CUSPARSE_DIAG_TYPE_NON_UNIT);
+  cusparseSetMatType (descrL,CUSPARSE_MATRIX_TYPE_GENERAL);
+
+  // ==============================================
+  // step 2: query how much memory used in csrsv2, and allocate the buffer
+  // pBuffer returned by cudaMalloc is automatically aligned to 128 bytes.
+  int pBufferSize;
+  void *pBufferL = 0;
+  cusparseOperation_t transL = (col_majorL ? CUSPARSE_OPERATION_TRANSPOSE : CUSPARSE_OPERATION_NON_TRANSPOSE);
+  if (std::is_same<scalar_t, double>::value == true) {
```

Comment:
`== true` is superfluous here, since the type of `std::is_same<U, V>::value` is `bool`.

Reply:
We removed "== true".